contentprovider.go

```diff
@@ -453,19 +452,24 @@ func (nls newlines) getLines(data []byte, low, high int) []byte {
 }
 
 const (
-	// TODO - how to scale this relative to rank?
+	// Query-dependent scoring signals
 	scorePartialWordMatch = 50.0
 	scoreWordMatch        = 500.0
 	scoreBase             = 7000.0
 	scorePartialBase      = 4000.0
 	scoreSymbol           = 7000.0
 	scorePartialSymbol    = 4000.0
 	scoreKindMatch        = 100.0
-	scoreFactorAtomMatch  = 400.0
-	scoreShardRankFactor  = 20.0
-	scoreFileOrderFactor  = 10.0
-	scoreLineOrderFactor  = 1.0
 	scoreRepetitionFactor = 1.0
+	scoreFactorAtomMatch  = 400.0
+
+	// File-only scoring signals
+	scoreFileRankFactor  = 9000.0
```

Comment:
`scoreFileRankFactor` is the only new variable, everything else was just rearranged (we maintain the old behavior exactly when UseDocumentRanks is disabled).

Explaining my approach here... In general there are two classes of scoring signals: query-document signals, which capture how well the document matches the query, and static document signals, which measure the importance of a document. When score signals are bounded and meaningful, it's super common to just sum them together with weights.

The query-dependent signals are bounded at ~9000 (`scoreWordMatch + scoreSymbol + scoreKindMatch * 10 + scoreFactorAtomMatch). So for now I set `scoreFileRankFactor = 9000` to give it equal weight. I'll test this on real data though using the test parameter `DocumentRanksWeight` to select a good default. I didn't touch the other static file signals (scoreFileOrderFactor and scoreShardRankFactor), their scoring contribution is relatively very small.

Reply:
It might be helpful to add this or something similar as comment in the code for future travellers. WDYT?

Reply:
Good point, will add a comment!